lib/rouge/lexers/python.rb

```diff
@@ -40,7 +40,7 @@ def self.builtins
       end
 
       def self.builtins_pseudo
-        @builtins_pseudo ||= %w(self None Ellipsis NotImplemented False True)
```

Comment:
So I kinda came at this from two sides:
1. In Python, `self` is not actually special. It isn't a keyword that has any special behavior in and of itself. It is definitely not the same as the other keywords from `@builtins_pseudo`. [This](https://docs.python.org/3/library/constants.html) page from the Python docs shows all of the built-in constants and `self` isn't one of them. In Python, `self` is just a local variable and its use is just customary, not necessary.
2. When I was working on the rouge GitHub theme, I was comparing how Python is highlighted on GitHub vs rouge. The GitHub highlighter doesn't highlight `self`:
```python
def __init__(self, input_dim: list, output_dim: int, **kwargs):
    super(AverageEmbedding, self).__init__(**kwargs)
    self.input_dim = input_dim
    self.output_dim = output_dim

@abstractmethod
def evaluate(self, *args, **kwargs):
    raise NotImplementedError

def _get_metadata(self, input_samples: list[InputSample]) -> Tuple[str, str, int]:
    project_name, id = self._get_info()
```

So based on both of those points, I removed it from `@builtins_pseudo` because I thought it better to not style it. Sorry, for not including it in the original message of the PR, that was a mistake.

Reply:
Fair point. Thank you for the explanation @dunkmann00 👍🏼 

For context, I believe that we inherit `self` highlight from [Pygments](https://github.com/pygments/pygments/blob/9d561d47ad32e5290708d7a290bac45d54ab9688/pygments/lexers/python.py#L248) which is used in the Python docs site.

Reply:
Oo interesting. I think I checked and saw that Pygments was highlighting `self` as well so I figured that was where it came from, but I didn't realize the Python docs site uses it. Interesting.